api/v1beta1/spifilecontentrequest_types.go

```diff
@@ -0,0 +1,70 @@
+//
+// Copyright (c) 2021 Red Hat, Inc.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1beta1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+type SPIFileContentRequestSpec struct {
+	FilePath string `json:"filePath"`
+	RepoUrl  string `json:"repoUrl"`
+	Ref      string `json:"ref,omitempty"`
+}
+
+type SPIFileContentRequestStatus struct {
+	Phase             SPIFileContentRequestPhase `json:"phase"`
+	LinkedBindingName string                     `json:"linkedBindingName"`
+	ErrorMessage      string                     `json:"errorMessage,omitempty"`
+	OAuthUrl          string                     `json:"oAuthUrl,omitempty"`
+	UploadUrl         string                     `json:"uploadUrl,omitempty"`
+	Content           string                     `json:"content,omitempty"`
+	ContentEncoding   string                     `json:"contentEncoding,omitempty"`
+	SHA               string                     `json:"sha,omitempty"`
```

Comment:
`SHA` of what? The file? The commit hash? How and when it is computed? Do we obtain it from the SP or do we compute it ourselves when downloaded? Is it verified? So many questions that the code comments could answer 😃

Reply:
since it is computed differently for different providers. i tend to remove it at all...